boa/src/builtins/set/mod.rs

```diff
@@ -0,0 +1,420 @@
+//! This module implements the global `Set` objest.
+//!
+//! The JavaScript `Set` class is a global object that is used in the construction of sets; which
+//! are high-level, collections of values.
+//!
+//! More information:
+//!  - [ECMAScript reference][spec]
+//!  - [MDN documentation][mdn]
+//!
+//! [spec]: https://tc39.es/ecma262/#sec-set-objects
+//! [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set
+
+use crate::{
+    builtins::{iterable::get_iterator, BuiltIn},
+    object::{ConstructorBuilder, FunctionBuilder, ObjectData, PROTOTYPE},
+    property::Attribute,
+    symbol::WellKnownSymbols,
+    BoaProfiler, Context, Result, Value,
+};
+use ordered_set::OrderedSet;
+
+pub mod set_iterator;
+use set_iterator::{SetIterationKind, SetIterator};
+
+pub mod ordered_set;
+#[cfg(test)]
+mod tests;
+
+#[derive(Debug, Clone)]
+pub(crate) struct Set(OrderedSet<Value>);
+
+impl BuiltIn for Set {
+    const NAME: &'static str = "Set";
+
+    fn attribute() -> Attribute {
+        Attribute::WRITABLE | Attribute::NON_ENUMERABLE | Attribute::CONFIGURABLE
+    }
+
+    fn init(context: &mut Context) -> (&'static str, Value, Attribute) {
+        let _timer = BoaProfiler::global().start_event(Self::NAME, "init");
+
+        let species = WellKnownSymbols::species();
+
+        let species_getter = FunctionBuilder::new(context, Self::species_getter)
+            .callable(true)
+            .constructable(false)
+            .name("get [Symbol.species]")
+            .build();
+
+        let size_getter = FunctionBuilder::new(context, Self::size_getter)
+            .callable(true)
+            .constructable(false)
+            .name("get size")
+            .build();
+
+        let iterator_symbol = WellKnownSymbols::iterator();
+
+        let to_string_tag = WellKnownSymbols::to_string_tag();
+
+        let values_function = FunctionBuilder::new(context, Self::values)
+            .name("values")
+            .length(0)
+            .callable(true)
+            .constructable(false)
+            .build();
+
+        let set_object = ConstructorBuilder::new(context, Self::constructor)
```

Comment:
```suggestion
        let set_object = ConstructorBuilder::with_standard_object(
            context,
            Self::constructor,
            context.standard_objects().set_object().clone(),
        )
```
Here we should init the global Set object